app/models/goal.py

```diff
@@ -1,6 +1,22 @@
 from flask import current_app
 from app import db
+from .task import Task
 
 
 class Goal(db.Model):
-    goal_id = db.Column(db.Integer, primary_key=True)
+    id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title = db.Column(db.String, nullable=False)
+    tasks = db.relationship('Task', back_populates='goal')
+
+    def to_dict(self):
+        return {
+            'id': self.id,
+            'title': self.title
+        }
+    
+    def to_dict_with_tasks(self):
+        return {
+            'id': self.id,
+            'title': self.title,
+            'tasks': [task.to_dict_with_goal() for task in self.tasks]
+        }
```

Comment:
Nice job on these! As an alternative, you could have a single function that only includes the tasks attribute if there are no tasks assigned to the goal.